docs/design/configs.md

```diff
@@ -0,0 +1,231 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named ``ConfigData`` stored inside ``etcd`` that
+stores command-line flags, environment variables, and configuration files used 
```

Comment:
@ncdc ack

@smarterclayton they will be string key value pairs regardless of the use. For example,
```json
{
  "data": {
    "MY_ENV": "some var",
    "config.yaml": "FILE_CONTENTS",
    ...
  }
}
```
The difference is, they will either be made available to component as environment variables, or as a mounted volume containing keys as file name and values as the file contents as I understood from the previous discussions since they can be used either way (depends on the consumer of ConfigData).

The proposal initially contained different fields for each type (``Flags``, ``EnvVars``, etc.)